### DISCUSSION TITLE
Seemingly random behaviour of 'Depends' while accessing a function during API call.

### DISCUSSION THREAD

**Discussion by space219** [fastapi/fastapi#5810] [Questions]
Hi, 
I have been working with FastAPI for managing websocket connections, and I am saving the websocket connections in a dictionary. However, whenever I try to access this active connection array, there are seemingly random instances wherein this array returns empty while at other times, it outputs the correct connection instances. 

My workflow is derived from the websocket tutorial on the FastAPI website.  I am using an external library that manages this websocket connection to perform various functionalities (defined as `ServerPoint` in the `ConnectionManager` class).

```python
'''Main Websocket Route'''

@app.websocket("/{obj_id}")
async def websocket_endpoint(websocket: WebSocket, obj_id: str, active_clients = Depends(get_active_clients)):
    await websocket.accept()
    try:
        await ws_manager.connect(websocket, obj_id, active_clients)
    except WebSocketDisconnect:
        await websocket.close() 

'''Connection Manager'''
class ConnectionManager:
 
    async def connect(self, websocket: WebSocket, obj_id: str, active_connections):
        client = ServerPoint(obj_id, websocket)
        try:        
            active_connections[cp_id] = client
            await client.start()

        except WebSocketDisconnect:
            active_connections.pop(cp_id)
            
ws_manager = ConnectionManager()
active_clients: Dict[str, ServerPoint] = {}

'''Dependency function'''
async def get_active_clients() -> Dict[str, ServerPoint]:
    return active_clients
```

The major issue arises wherein I have defined an additional HTTP route which looks to perform certain operations defined in the `ServerPoint` class on an individual connection stored in the dictionary, but the behaviour of the `Depends(get_active_clients)` is pretty erratic with it returning an empty dictionary on some occasions and returning full instances on another. 

A sample testing route that I created to check it is defined below. It is difficult to reproduce the error as I am not able to understand any real pattern to the behaviour. 

```python
@app.get("/test-active)
def test_dict_keys(active_clients: Dict[str, ServerPoint] = Depends(get_active_clients)):
    clients = list(active_clients.keys())

    return clients 
```
It would be greatly helpful if someone could guide me to solve this issue and be able to generate a robust workflow.

Thanks

**Comment by YuriiMotov** [comment]
Duplicate: https://github.com/fastapi/fastapi/discussions/6053
